### ISSUE TITLE
libtorrent doesn't call remove() in POSIX wrapper

### ISSUE THREAD

**Issue by proninyaroslav** [aldenml/libtorrent4j#45]
Hi. I tried to implement a POSIX wrapper, as done in `frostwire`. The only difference is that I use a fake path to access `Storage Access Framework` (which I decode in the wrapper). 
In general, this works fine, except that it's not possible to delete a file on the SD card that is accessible via the fake path. If the path is real (for example, `/storage/emulated/0`) then the files are deleted. 
I checked and it turned out that the `remove(String path)` method isn't called by `libtorrent` in the POSIX wrapper. I assume `libtorrent` is trying to delete the file bypassing the POSIX wrapper, but since I use the fake path for `SAF`, it fails. Perhaps this is a bug?

**Comment by aldenml** [comment]
sounds like a bug to me, maybe there was a change in the `libtorrent` source code that I didn't take into account

**Comment by proninyaroslav** [comment]
Do you have any news? This issue is quite important compared to other POSIX wrapper problems, without this I can’t make a new release of the app :(

**Comment by aldenml** [comment]
Hi @proninyaroslav, still nothing, but this is on my list

**Comment by proninyaroslav** [comment]
`1.2.2.0` still has this problem.

**Comment by proninyaroslav** [comment]
I made `strace` and noticed that `libtorrent` doesn't use the `remove()` system call, but `unlinkat()` to delete the file. Therefore, I can't catch file removing in `posix_wrapper`:
```
unlinkat(AT_FDCWD, "/saf_root(com.android.externalstorage.documents/tree/1018-230F%3A);/CCleaner Professional Plus v5.25.0.5902 x86-x64 Setup + CRACK.7z", 0) = -1 ENOENT (No such file or directory)
unlinkat(AT_FDCWD, "/saf_root(com.android.externalstorage.documents/tree/1018-230F%3A);/.26e8d03aff486d164e262d0a09cadbe8fc4f71a9.parts", 0) = -1 ENOENT (No such file or directory)
```

**Comment by proninyaroslav** [comment]
@aldenml  I think you need to add this call to `posix_wrapper`.

**Comment by proninyaroslav** [comment]
On Android 6.0, I have a different situation:
```
fstatat64(AT_FDCWD, "/saf_root(com.android.externalstorage.documents/tree/10F4-1006%3A);/F-Secure Freedome VPN 2.32.6293 RePack by elchupacabra/", 0x8f7fb458, AT_SYMLINK_NOFOLLOW) = -1 ENOENT (No such file or directory)
```
Instead of `unlinkat()` calls, I have many `fstatat64()` calls, which, of course, are unsuccessful.

**Comment by aldenml** [comment]
@proninyaroslav thank you very much for this information, it is really helpful. I will not be able to work in this for a few more days.

**Comment by proninyaroslav** [comment]
@aldenml If you need other information or strace's from other Android versions, let me know.

**Comment by aldenml** [comment]
@proninyaroslav which version of `libtorrent4j` are you using for this strace?

**Comment by proninyaroslav** [comment]
@aldenml `1.2.3.0`

**Comment by aldenml** [comment]
that's strange, I don't use the weak symbols hack in this version, it should be something else

**Comment by proninyaroslav** [comment]
That’s all I could see useful in strace. The only fact is that the POSIX wrapper `remove()` method is not called.

**Comment by proninyaroslav** [comment]
Fixed in 1.3.0-alpha-5
